workflow/controller/controller.go

```diff
@@ -553,37 +554,66 @@ func (wfc *WorkflowController) workflowGarbageCollector(stopCh <-chan struct{})
 					log.WithField("err", err).Error("Failed to list old offloaded nodes")
 					continue
 				}
-				if len(oldRecords) == 0 {
-					log.Info("Zero old offloads, nothing to do")
-					continue
-				}
-				// get every lives workflow (1000s) into a map
-				liveOffloadNodeStatusVersions := make(map[types.UID]string)
-				workflows, err := util.NewWorkflowLister(wfc.wfInformer).List()
-				if err != nil {
-					log.WithField("err", err).Error("Failed to list incomplete workflows")
-					continue
+				lenOldOffloads := 0
```

Comment:
I think you can remove this diagnostics output - just more code that needs to be tested and. maintained.